phoenixRest/models/core/friendship.py

```diff
@@ -0,0 +1,46 @@
+from sqlalchemy import (
+    Column,
+    ForeignKey,
+    DateTime
+)
+from sqlalchemy.dialects.postgresql import UUID
+from sqlalchemy.orm import relationship
+
+from phoenixRest.models import Base
+
+import uuid
+
+from datetime import datetime
+
+class Friendship(Base):
+    __tablename__ = "friendship"
+    uuid = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4, unique=True, nullable=False)
+    
+    source_user_uuid = (Column(UUID(as_uuid=True), ForeignKey("user.uuid"), nullable=False))
+    source_user = relationship("User", foreign_keys=[source_user_uuid])
+    
+    recipient_user_uuid = Column(UUID(as_uuid=True), ForeignKey("user.uuid"), nullable=False)
+    recipient_user = relationship("User", foreign_keys=[recipient_user_uuid])
+    
+    accepted = Column(DateTime, nullable=True)
+    
+    revoked = Column(DateTime, nullable=True)
+    
+    created = Column(DateTime, nullable=False)
+    
+    def __init__(self, source_user, recipient_user, accepted, revoked):
```

Comment:
As i wrote in the view handler file, if you are planning on passing `None` to the constructor always, i'd rather you rely on the defaults - `accepted` and `revoked` will both be True if you don't set them - why do it explicitly?

Reply:
Ok so if i understand correctly: Since both the accepted and revoked columns are defined as nullable, they will automatically get the default value of None, thus making the constructor arguments unnecessary.

Reply:
Does that mean i can remove self.accepted and self.revoked aswell?

Reply:
Correct